File: Java/Medium/CompileError/number-of-sub-arrays-of-size-k-and-average-greater-than-or-equal-to-threshold_8.java
class Solution {
    public int numOfSubarrays(int[] arr, int k, int threshold) {
        
        int sum = 0;
        for (int i = 0; i < k; i++) {
            sum += arr[i];
        }
        int result = 0;
        if (sum >= threshold * k) {
            result++;
        }
        for (int i = k; i < arr.length; i++) {
            sum += arr[i] - arr[i - k];
            if (sum >= threshold * k) {
                result++;
            }
        } threshold) {
                count++;
            }
        }
        return count;
    }
}

File: Rust/Medium/Accepted/number-of-sub-arrays-of-size-k-and-average-greater-than-or-equal-to-threshold_8.rs
impl Solution {
    pub fn num_of_subarrays(arr: Vec<i32>, k: i32, threshold: i32) -> i32 {
        
        let k = k as usize;
        let threshold = threshold as i32;
        let mut sum = 0;
        for i in 0..k {
            sum += arr[i];
        }
        let mut num = 0;
        if sum / k as i32 >= threshold {
            num += 1;
        }
        for i in k..arr.len() {
            sum = sum - arr[i - k] + arr[i];
            if sum / k as i32 >= threshold {
                num += 1;
            }
        }
        num
    }
}